tests/lib/fakeinstaller/main.go

```diff
@@ -82,11 +82,48 @@ func runMntFor(label string) string {
 	return filepath.Join(dirs.GlobalRootDir, "/run/fakeinstaller-mnt/", label)
 }
 
-func postSystemsInstallSetupStorageEncryption(details *client.SystemDetails) error {
-	// TODO: check details.StorageEncryption and call POST
-	// /systems/<seed-label> with "action":"install" and
-	// "step":"setup-storage-encryption"
-	return nil
+func postSystemsInstallSetupStorageEncryption(cli *client.Client,
+	details *client.SystemDetails, bootDevice string,
+	onDiskParts []gadget.OnDiskStructure) (*client.Change, error) {
+
+	// We are modifiying the details struct here
+	for _, gadgetVol := range details.Volumes {
+		for i := range gadgetVol.Structure {
+			switch gadgetVol.Structure[i].Role {
+			case "system-save", "system-data":
+				// only roles for which we will want encryption
+			default:
+				continue
+			}
+			for _, part := range onDiskParts {
+				if part.Name == gadgetVol.Structure[i].Name {
+					gadgetVol.Structure[i].UnencryptedDevice = part.Node
```

Comment:
yes it seems this could set Device instead as what is what is clear from usage context